### PR TITLE
Fix mp serialization for integer nn.Parameter on CUDA

### LINKED ISSUE #56342: Multiprocessing with tensors on GPU: "Only Tensors of floating point dtype can require gradients"
## 🐛 Bug

Actually, the bug is almost the same as described [here](https://github.com/pytorch/pytorch/issues/17345), only for CUDA tensors, not CPU.

## To Reproduce

```python
import torch
import torch.multiprocessing as mp
import torch.nn as nn

class Net(nn.Module):
    def __init__(self):
        super().__init__()
        self.i = nn.Parameter(torch.tensor(0), requires_grad=False)

def run(rank, net):
    print('start run rank:', rank)

if __name__ == "__main__":
    net = Net()

    net.to('cuda:0')  # Here is the difference from the example in the linked PR

    net.share_memory()
    c = mp.spawn(run, args=[net], nprocs=1, join=False)
    c.join()
```

## Expected behavior

No error.

## Environment

```
PyTorch version: 1.7.0
Is debug build: True
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.3 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.7 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti

Nvidia driver version: 450.102.04
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.6.3
[pip3] numpy==1.19.2
[pip3] segmentation-models-pytorch==0.1.3
[pip3] torch==1.7.0
[pip3] torchaudio==0.7.0
[pip3] torchvision==0.8.1
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.1.243             h6bb024c_0  
[conda] efficientnet-pytorch      0.6.3                    pypi_0    pypi
[conda] mkl                       2020.2                      256  
[conda] mkl-service               2.3.0            py37he904b0f_0  
[conda] mkl_fft                   1.2.0            py37h23d657b_0  
[conda] mkl_random                1.1.1            py37h0573a6f_0  
[conda] numpy                     1.19.2           py37h54aff64_0  
[conda] numpy-base                1.19.2           py37hfa32c7d_0  
[conda] segmentation-models-pytorch 0.1.3                    pypi_0    pypi
[conda] torch                     1.7.0                    pypi_0    pypi
[conda] torchaudio                0.7.0                    pypi_0    pypi
[conda] torchvision               0.8.1                    pypi_0    pypi
```

## Additional context

Is suggest fixing the problem the same way as for the [Issue 17345](https://github.com/pytorch/pytorch/issues/17345): an `if` in the `rebuild_cuda_tensor` function:
https://github.com/pytorch/pytorch/blob/master/torch/multiprocessing/reductions.py#L127

If the idea is OK, I would be happy to make a small PR with the fix.

cc @ngimel

### PR DESCRIPTION
**Pull request by Alvant**
Fixes #56342

### CODE REVIEW

**Review by ngimel — APPROVED** [review]
Thanks for the fix

File: test/test_multiprocessing.py [hunk 1]
```diff
@@ -833,12 +833,21 @@ def test_cuda_parameter_sharing(self):
     @unittest.skipIf(NO_MULTIPROCESSING_SPAWN, "Disabled for environments that \
                      don't support multiprocessing with spawn start method")
     def test_integer_parameter_serialization(self):
-        iparam = torch.nn.Parameter(torch.tensor(0, dtype=torch.int64), requires_grad=False)
+        for device in ['cpu', 'cuda']:
```
Comment (ngimel) [inline test/test_multiprocessing.py:836]: Test errors are real, you can skip this test if cuda ipc is not available, like the tests above. Also, nit, prefer using tuples ('cpu', 'cuda') over lists.

Comment (Alvant) [reply test/test_multiprocessing.py]: @ngimel oops! my mistake. Thanks for the advice.

I split the test in two: for CPU and for CUDA. The first one is going to be run regardless of the availability of CUDA IPC. The second test is skipped if CUDA is not available.
